### DISCUSSION TITLE
dpgen: How to assign different INCAR paramaters for each first principle (vasp) calculation?

### DISCUSSION THREAD

**Discussion by xvcheng** [deepmodeling/dpgen#1426] [Q&A]
I'd like to build a deep potential model to pridict the energy of the charegd systems with different defects configuration using **dpgen**.  In vasp, `NELECT` should be specified in charged systems. 

The init data inclues bulk and charged systems (with different `NELECT` for different defects). 

In dpgen, `Model_devi` will find the some poorly-fitted structures for **vasp** calculation.  But how can I specify the differnet `NELECT` parameters  for each fp job?  Because `"fp_incar"` only accpect one INCAR file and different systems have different `NELECT` .

Is there any ways in dpgen to achieve this funciton?

One way I thought of is to run a user-defined python script before each submission of fp job by **slurm**. The script is used to check the atoms and add the `NELECT` to the INCAR. But I don't know is there any parameters in dpgen can achieve this.

Any help is appreciated!

**Comment by xvcheng** [comment]
Can I put the python script into `source_list` ?

**Comment by njzjz** [comment]
It looks to me you could set the command to something like `python /path/to/some.py && vasp_std`, so the script will always run before each vasp run.

**Reply by xvcheng** [reply]
Thanks! I'll try it now.
